nearcore/src/runtime/mod.rs

```diff
@@ -1413,14 +1441,17 @@ impl RuntimeAdapter for NightshadeRuntime {
         part_id: u64,
         num_parts: u64,
         data: &[u8],
+        epoch_id: &EpochId,
     ) -> Result<(), Error> {
         let part = BorshDeserialize::try_from_slice(data)
             .expect("Part was already validated earlier, so could never fail here");
-        let trie_changes = Trie::apply_state_part(&state_root, part_id, num_parts, part)
-            .expect("combine_state_parts is guaranteed to succeed when each part is valid");
+        let (trie_changes, contract_codes) =
+            Trie::apply_state_part(&state_root, part_id, num_parts, part)
```

Comment:
I wonder why we even bother with Result here if we always just panic on error

Reply:
Could you tell more about what can be improved here?
I see that even on the `Trie::apply_state_part` level `StorageError` can be returned at many places, so it makes sense to process it here in one place.

Reply:
maybe it would be better to have a wrapper function then?